### PR TITLE
#748 - Added a command line option to specify the log files name.

### PR DESCRIPTION
**Pull request by katbyte**
Fix for issue #748, added a command line option to specify the log files name.

### CODE REVIEW

**Comment by tombruijn** [comment]
Hi @katbyte thanks for spotting the weird docs, but reading further I see there's already a `--logfile` option which enables the log file. If we add another option named `--log-file` I'm afraid it will be quite confusing what which one does.

**Comment by katbyte** [comment]
Good point. I renamed the option to --log-filename and updated the variables in the code to reflect the name change.

**Comment by tombruijn** [comment]
Hi @katbyte, sorry for the delay. Thanks for the update, looks good

I think we only need tests for this cli option and the logfile class setting. Can you add them?

File: lib/backup/logger/logfile.rb [hunk 1]
```diff
@@ -102,8 +107,8 @@ def setup_logfile
           path = File.join(Backup::Config.root_path, path)
         end
         FileUtils.mkdir_p(path)
-        log_file = @options.log_file || 'backup.log'
-        path = File.join(path, log_file)
+        log_filename = @options.log_filename.strip.empty? ? 'backup.log' : @options.log_filename
```
Comment (tombruijn) [inline lib/backup/logger/logfile.rb]: is `.strip` necessary here? The spaces are already stripped from the string when it's set in the `log_filename` method

**Comment by katbyte** [comment]
Sure thing, i'll try and get the tests written over the weekend/early next week

File: lib/backup/logger/logfile.rb (hunk 1 above)
Comment (nickcluc) [inline lib/backup/logger/logfile.rb]: Hey! Not the author but it seems as though you are right. options.log_filename will already be stripped by the time it makes it to this point! I could branch from here and make the change to get this merged if need be!

**Pull request closed without merging**